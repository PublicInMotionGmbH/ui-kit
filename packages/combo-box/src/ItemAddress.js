import React from 'react'
import PropTypes from 'prop-types'

import { buildClassName, prefix } from '@talixo/shared'

/**
 * Component which represents dropdown.
 *
 * @param {object} props
 * @param {string} [props.className]
 * @param {object} [props.item]
 * @param {object} [props.style]
 * @returns {React.Element}
 */
const ItemAddress = props => {
  const { item, className, style } = props

  const clsName = buildClassName('item-address', className)
  const iconClsName = buildClassName([ 'item-address', 'icon' ], 'material-icons')

  return (
    <div className={clsName} style={style}>
      <i className={iconClsName}>
        {item.icon}
      </i>
      <span className={prefix('item-address', 'place')}>{item.place}</span>
      <span className={prefix('item-address', 'address')}>{item.address}</span>
    </div>
  )
}

ItemAddress.propTypes = {
  /** Additional class name */
  className: PropTypes.string,

  /** Displayed item */
  item: PropTypes.object,

  /** Style object */
  style: PropTypes.object
}

ItemAddress.defaultProps = {}

export default ItemAddress
