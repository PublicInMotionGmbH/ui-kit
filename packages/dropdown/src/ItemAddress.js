import React from 'react'
import cls from 'classnames'
import PropTypes from 'prop-types'

import { prefix } from '@talixo/shared'

const moduleName = prefix('item-address')

/**
 * Component which represents dropdown.
 *
 * @param {object} props
 * @param {string} [props.className]
 * @param {object} [props.item]
 * @param {object} [props.style]
 * @returns {React.Element}
 */
const ItemAddress = props => {
  const { item, className, style } = props
  return (
    <div className={cls(moduleName, className)} style={style}>
      <i className={`material-icons ${moduleName}__icon`}>
        {item.icon}
      </i>
      <span className={`${moduleName}__place`}>{item.place}</span>
      <span className={`${moduleName}__address`}>{item.address}</span>
    </div>
  )
}

ItemAddress.propTypes = {
  /** Additional class name */
  className: PropTypes.string,

  /** Displayed item */
  item: PropTypes.object,

  /** Style object */
  style: PropTypes.object
}

ItemAddress.defaultProps = {}

export default ItemAddress
