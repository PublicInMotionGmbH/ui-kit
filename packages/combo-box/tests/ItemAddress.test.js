import React from 'react'
import ItemAddress from '../src/ItemAddress'
import { shallow } from 'enzyme'

describe('<ItemAddress />', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<ItemAddress item={{ icon: 'credit_card', place: 'Hotel', address: 'Berlin' }} />)

    expect(wrapper).toMatchSnapshot()
  })
})
