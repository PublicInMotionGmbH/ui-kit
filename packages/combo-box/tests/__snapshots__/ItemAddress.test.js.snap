// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ItemAddress /> renders correctly 1`] = `
<div
  className="talixo-item-address"
>
  <Icon
    className="talixo-item-address__icon material-icons"
    name="credit_card"
  />
  <span
    className="talixo-item-address__place"
  >
    Hotel
  </span>
  <span
    className="talixo-item-address__address"
  >
    Berlin
  </span>
</div>
`;
